ignite/engine/events.py

```diff
@@ -4,50 +4,68 @@
 from enum import Enum
 import numbers
 import weakref
+from copy import deepcopy
 
 from ignite.engine.utils import _check_signature
 
-
 __all__ = [
     'Events',
     'State'
 ]
 
 
-class EventWithFilter:
+class CallableEventWithFilter:
+    """Single Event containing a filter, specifying whether the event should
+    be run at the current event (if the event type is correct)
 
-    def __init__(self, event: Any, filter: Callable):
-        if not callable(filter):
-            raise TypeError("Argument filter should be callable")
-        self.event = event
-        self.filter = filter
+    Args:
+        value (str): The value (will only be set if not yet set by other parent class). Necessary for usage with
+            :class:`enum.Enum`
+        event_filter (callable): A function taking the engine and the current event value as input and returning a
+            boolean to indicate whether this event should be executed. Defaults to None, which will result to a
+            function that always returns `True`
 
-    def __str__(self) -> str:
-        return "<%s event=%s, filter=%r>" % (self.__class__.__name__, self.event, self.filter)
+    """
 
+    def __init__(self, value: str, event_filter: Optional[Callable] = None):
+        self._filter = lambda _, __: True
+        self.filter = event_filter
```

Comment:
This line and above both look a bit strange. Especially, that `filter` then becomes a property...

Reply:
Basically the property wraps the `self._filter` attribute, where the setter handles the None argument and makes it a function returning always `True`. This also answers your question below.

Reply:
Yes, I see now. But creation of setter just for `self.filter = event_filter` is a bit overkill :) 
My below question is about whether we need to reset event filter dynamically ? I think we should not

Reply:
Otherwise we need to check for `None` all the time, so I found this to be more elegant. But I agree, that we might not want to change it dynamically, so I will just hardcode the check where needed.